### PR TITLE
send platform notification for courses with 7 days of inactivity afte…

### PR DESCRIPTION
**Pull request by muak50**
…r course open date

### CODE REVIEW

File: common/djangoapps/student/views.py [hunk 1]
```diff
@@ -839,6 +852,43 @@ def dashboard(request):
...
+
+        course = CourseOverview.objects.get(id=user_course.course_id)
+        overview_courses.append(course)
+        course_start_date = get_course_open_date(course).date()
+        log.info('Course start date %s', course_start_date)
+
+        delta_date = today - course_start_date
+        log.info('Days passed since course started %s', delta_date.days)
+
+        if delta_date.days >= DAYS_TO_SEND_EMAIL:
```
Comment (emraan004) [inline common/djangoapps/student/views.py]: let's change the variable name

File: common/djangoapps/student/views.py [hunk 2]
```diff
@@ -839,6 +852,43 @@ def dashboard(request):
...
+    active_courses = []
+    overview_courses = []
+
+    for user_course in all_user_courses:
+
+        today = datetime.datetime.now(utc).date()
+        log.info('Today date %s', today)
+
+        course = CourseOverview.objects.get(id=user_course.course_id)
+        overview_courses.append(course)
```
Comment (emraan004) [inline common/djangoapps/student/views.py]: let's append in list when criteria is met

File: common/djangoapps/student/views.py [hunk 3]
```diff
@@ -839,6 +852,43 @@ def dashboard(request):
...
+
+            for mod_entry in modules:
+                # Verifying if mod_entry is after Course Open date
+                if course_start_date < mod_entry.created.date():
+                    active_courses.append(course)
+                    break
+        else:
+            overview_courses.remove(course)
+
+    non_active_courses = [course for course in overview_courses if course not in active_courses]
```
Comment (emraan004) [inline common/djangoapps/student/views.py]: we might not need this list

File: common/djangoapps/student/views.py [hunk 4]
```diff
@@ -791,6 +796,13 @@ def dashboard(request):
     else:
         redirect_message = ''
 
+    # Importing get_course_first_chapter_link at global space makes a circular dependency importing problem.
+    from openedx.core.djangoapps.timed_notification.core import get_course_first_chapter_link
+    first_non_active_course = {}
+    if get_non_active_course(user).__len__() > 0:
```
Comment (emraan004) [inline common/djangoapps/student/views.py]: let's use len(list)

**Comment by muak50** [comment]
Merging for testing... Procedure will be reviewed after testing.

**Pull request merged**